streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -297,7 +308,23 @@ public synchronized final TopologyBuilder setApplicationId(String applicationId)
      * @return this builder instance so methods can be chained together; never null
      */
     public synchronized final TopologyBuilder addSource(String name, String... topics) {
-        return addSource(name, (Deserializer) null, (Deserializer) null, topics);
+        return addSource(name, null, (Deserializer) null, (Deserializer) null, topics);
+    }
+
+    /**
+     * Add a new source that consumes the named topics and forward the records to child processor and/or sink nodes.
+     * The source will use the {@link org.apache.kafka.streams.StreamsConfig#KEY_SERDE_CLASS_CONFIG default key deserializer} and
+     * {@link org.apache.kafka.streams.StreamsConfig#VALUE_SERDE_CLASS_CONFIG default value deserializer} specified in the
+     * {@link org.apache.kafka.streams.StreamsConfig stream configuration}.
+     *
+     * @param name the unique name of the source used to reference this node when
+     * {@link #addProcessor(String, ProcessorSupplier, String...) adding processor children}.
+     * @param offsetReset the offset reset to use for this source if no committed offsets found; acceptable values earliest or latest
+     * @param topics the name of one or more Kafka topics that this source is to consume
+     * @return this builder instance so methods can be chained together; never null
+     */
+    public synchronized final TopologyBuilder addSource(String name, KStreamBuilder.AutoOffsetReset offsetReset, String... topics) {
```

Comment:
Maybe put offsetRest before `String name` to be consistent with the `KStreamBuilder`?